src/app/src/components/SubmissionDetail.js

```diff
@@ -0,0 +1,199 @@
+import {
+    Box,
+    Button,
+    Checkbox,
+    Divider,
+    Flex,
+    Grid,
+    GridItem,
+    Heading,
+    HStack,
+    Icon,
+    IconButton,
+    Spacer,
+    Text,
+} from '@chakra-ui/react';
+import {
+    ArrowLeftIcon,
+    ChatAltIcon,
+    DownloadIcon,
+    MailIcon,
+    InformationCircleIcon,
+} from '@heroicons/react/outline';
+import { useNavigate, useParams } from 'react-router-dom';
+import { MapContainer } from 'react-leaflet';
+
+import ActivityLog from './ActivityLog';
+import { DefaultBasemap } from './Layers/Basemaps';
+import MapPanes from './MapPanes';
+import { SubmittedBadge } from './SubmissionBadges';
+
+import { MAP_CENTER, MAP_INITIAL_ZOOM } from '../constants';
+
+export default function SubmissionDetail() {
+    const navigate = useNavigate();
+    const params = useParams();
+
+    const submissionId = params.id;
+
+    const fetchSubmissionDetails = submissionId => {
+        return {
+            submissionName: 'Raleigh',
+            contactName: 'Jenny Smith',
+            contactPhone: '1-919-123-4567',
+            contactTitle: 'Water System Operator',
+            pwsId: '0392010',
+            utilityName: 'Raleigh City Of',
+            utilityAddress1: 'Division of Water Resources',
+            utilityAddress2: '1634 Mail Service Center',
+            utilityCity: 'Raleigh',
+            utilityState: 'NC',
+            utilityZip: '27699-1634',
+            sharingPreference: true,
+            status: 'submitted',
+        };
+    };
+
+    const {
+        submissionName,
+        contactName,
+        contactPhone,
+        contactTitle,
+        pwsId,
+        utilityName,
+        utilityAddress1,
+        utilityAddress2,
+        utilityCity,
+        utilityState,
+        utilityZip,
+        sharingPreference,
+    } = fetchSubmissionDetails(submissionId);
+
+    const cityStateZip = `${utilityCity}, ${utilityState} ${utilityZip}`;
+
+    const makeDataGrid = ({ initialMarginTop, title, data }) => {
```

Comment:
Is there a performance benefit to turning this arrow function into a module-level function? I know there would be in Python, but I'm new enough to React to not have an intuition for this.

Reply:
Decomposing into smaller components took care of this.